epinorm/norm.py

```diff
@@ -1,40 +1,138 @@
     "locality_osm_id",
-    "original_record_source",
-    "original_record_id",
+    "admin_level_1",
+    "admin_level_1_osm_id",
+
+    # "original_record_source",
+    # "original_record_id",
     "original_record_location_description",
 ]
 
+def get_translitaration_endonym(row):
+        """
+        this tries to transliterate the endonym of a row into latin characters.
+        If it fails, it returns an empty list
+        """
+
+        # try to tranlisterate the endonym
+        try:
+            transliteration = translit(row["endonym"], reversed=True)
+
+            # if the original language was russian, then it might have used the character ь which
+            # doesn't get transliterate properly (it gets into an apostrophe)
+            if "ь" in row["endonym"] and "'" in transliteration:
+                transliteration = transliteration.replace("'", "")
+
+            entry = {"name": transliteration, 
+                    "admin_level":row["admin_level"], 
+                    "osm_id": row["osm_id"]}
+            return [entry]
+
+        except (LanguageDetectionError, TypeError):
+            return []
+
+def get_admin_levels_table():
+    """
+    Retrieves the information in "countries.csv" and
+    "administrative_units.tsv", and returns a dictionary that maps a
+    country name (from "countries.csv") to a list of its administrative
+    levels.
+    Administrative levels are encoded as dictionaries with properties:
+    {
+        name,
+        admin_level,
+        osm_id
+    }
+    Also transliterates information from the administrative_units.tsv file.
+    """
+
+    # * create a mapping from country name to its two letter code
+    country_to_code = pd.read_csv(COUNTRIES_DATA, index_col="name")["alpha_2"].to_dict()
+    country_to_code.update(COUNTRIES_EXCEPTIONS) # add the exceptions
+
+    # * now map country code to its admin levels
+    code_to_admin = {}
+    for country_code, rows_country in pd.read_table(ADMIN_LEVELS_DATA).groupby("iso3166_1_code"):
+
+        code_to_admin[country_code] = []
+        for _, row in rows_country.iterrows():
+
+            new_entries = []
+
+            if isinstance(row["exonym"], str): # make sure the column contains a value
+                entry = {"name":row["exonym"],
+                         "admin_level":row["admin_level"], 
+                         "osm_id":row["osm_id"]}
+                new_entries.append(entry)
+
+            if isinstance(row["endonym"], str): # make sure the column contains a value
+                entry = {"name": row["endonym"], 
+                        "admin_level":row["admin_level"], 
+                        "osm_id": row["osm_id"]}
+                new_entries.append(entry)
+
+                new_entries += get_translitaration_endonym(row)
+
+            # some words are okay to be replaced, they are modifiers
+            replacements = {
+            "District" : "Region",
+            "Region" : "District",
+            }
```

Comment:
The insertion of "region" and "district" pairs can lead to wrong results. It is possible for a region and a district to share the same name, because the two units are at different administrative levels. The idea behind the addition of synonyms is to accommodate naming variations for the same administrative units. For example, there are records in GenBank referring to "Omsk region", which corresponds to "Omsk Oblast" in OpenStreetMap. By providing a variant in which "oblast" is replaced with "region", we can correctly resolve the locations in those records.

```suggestion
            replacements = {
            "Oblast" : "Region",
            "Krai" : "Region",
            }
```